feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt

```diff
@@ -65,7 +66,12 @@ internal class AmazonBilling constructor(
     var connected = false
 
     override fun startConnection() {
-        purchasingServiceProvider.registerListener(applicationContext, this)
+        if (!Purchases.sharedInstance.observerMode) {
+            purchasingServiceProvider.registerListener(applicationContext, this)
+        } else {
+            LogIntent.AMAZON_ERROR(AmazonStrings.ERROR_OBSERVER_MODE_NOT_SUPPORTED)
```

Comment:
trying to put this in the builder proved trickier, and this log will be going away when we implement step 2 of the fix

Reply:
I think it's fine here. I would make sure we early return though, so we don't set `connected` to true. So maybe we can do:
```
if (Purchases.sharedInstance.observerMode) { 
    LogIntent.AMAZON_ERROR(AmazonStrings.ERROR_OBSERVER_MODE_NOT_SUPPORTED)
    return
}
purchasingServiceProvider.registerListener(applicationContext, this)
connected = true
```

Reply:
ah yes, woops! will do